src/main/scala/org/broadinstitute/hail/stats/package.scala

```diff
@@ -1,10 +1,63 @@
 package org.broadinstitute.hail
 
 import org.apache.commons.math3.distribution.HypergeometricDistribution
+import org.broadinstitute.hail.annotations.Annotation
+import org.broadinstitute.hail.expr.{TDouble, TInt, TStruct}
 import org.broadinstitute.hail.utils._
+import org.broadinstitute.hail.variant.Genotype
 
 package object stats {
 
+  object InfoScoreCombiner {
+    def signature = TStruct("score" -> TDouble, "nIncluded" -> TInt)
+  }
+
+  class InfoScoreCombiner {
+    var result = 0d
+    var expectedAlleleCount = 0d
+    var totalDosage = 0d
+    var nIncluded = 0
+
+    def expectedMean(dosage: Array[Double]): Double = dosage(1) + 2 * dosage(2)
+
+    def expectedVariance(dosage: Array[Double], mean: Double): Double = (dosage(1) + 4 * dosage(2)) - (mean * mean)
+
+    def merge(g: Genotype): InfoScoreCombiner = {
+      g.dosage.foreach { dx =>
+        val mean = expectedMean(dx)
```

Comment:
expectedMean is now called only once, you can put the computation back inline I think.